src/core/modules/StateEvaluator.ts

```diff
@@ -77,6 +77,7 @@ export class EvalStateResult<State> {
   constructor(
     readonly state: State,
     readonly validity: Record<string, boolean>,
+    readonly errorMessages: Record<string, string>,
```

Comment:
adding as a separate fields, as we need to retain backward compatibility with how currently `validity` is being stored - e.g. some contracts, that read other's contract state, might currently use the validity object and make boolean comparisons.